zebra/rt_netlink.c

```diff
@@ -3773,24 +3750,40 @@ static int netlink_ipneigh_change(struct nlmsghdr *h, int len, ns_id_t ns_id)
 	 * inteface
 	 * itself
 	 */
-	if (IS_ZEBRA_IF_VLAN(ifp)) {
+	if (IS_ZEBRA_IF_VLAN(ifp))
 		link_if = if_lookup_by_index_per_ns(zebra_ns_lookup(ns_id),
 						    zif->link_ifindex);
-		if (!link_if)
-			return 0;
-	} else if (IS_ZEBRA_IF_BRIDGE(ifp))
+	else if (IS_ZEBRA_IF_BRIDGE(ifp))
 		link_if = ifp;
-	else {
+	else
+		link_if = NULL;
+
+	if (tb[NDA_LLADDR]) {
+		l2_len = RTA_PAYLOAD(tb[NDA_LLADDR]);
+		l2_data = RTA_DATA(tb[NDA_LLADDR]);
+	} else {
+		l2_len = 0;
+		l2_data = NULL;
+	}
+
+	zsend_nhrp_neighbor_notify(
+		netlink_nbr_cmd_to_zclient(h->nlmsg_type), ifp, &ip,
+		netlink_nbr_entry_state_to_zclient(ndm->ndm_state), l2_data,
+		l2_len);
+
+	if (h->nlmsg_type == RTM_GETNEIGH)
```

Comment:
This code is not 1:1 functionality with the previous one. Suggestion: make this `if` into a `switch` and reject anything other than `RTM_NEWNEIGH`, `RTM_DELNEIGH`.

Here is what the previous remove code did (right above these lines addition):
```c
	if (h->nlmsg_type == RTM_NEWNEIGH)
		cmd = ZEBRA_NHRP_NEIGH_ADDED;
	else if (h->nlmsg_type == RTM_GETNEIGH)
		cmd = ZEBRA_NHRP_NEIGH_GET;
	else if (h->nlmsg_type == RTM_DELNEIGH)
		cmd = ZEBRA_NHRP_NEIGH_REMOVED;
	else {
		zlog_debug("%s(): unknown nlmsg type %u", __func__,
			   h->nlmsg_type);
		return 0;
	}
```